core/shared/src/main/scala/org/http4s/headers/X-Content-Type-Options.scala

```diff
@@ -20,12 +20,22 @@ package headers
 import cats.data.NonEmptyList
 import cats.parse.Parser
 import org.http4s.internal.parsing.CommonRules
-import org.typelevel.ci.{CIString, CIStringSyntax}
+import org.typelevel.ci.CIString
+import org.typelevel.ci.CIStringSyntax
+
+/** Response header that indicates whether the browser should be allowed to
+  * render the response as a downloadable file.
+  *
+  * [[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options X-Content-Type-Options]]
+  */
+sealed abstract case class `X-Content-Type-Options` private (value: CIString)
```

Comment:
Ah, yes, this `sealed abstract` is better than my suggestion while we still support Scala 2.12.

Reply:
on the second thought, `abstract case class` still generates the `unapply` method. AFAIR, there were concerns in regards to `unapply`-s in header models. Just because if it happens that their spec changes then it will be difficult to extend the model without breaking the bin-compat. Not sure if it is applicable in this particular case though. However, for now it looks like there's just a single value accepted – **nosniff**, so perhaps the extractor would not make a lot of sense anyway. Wdyt?

Reply:
Yes, a case class model is in bincompat trouble if they ever add more values.  I think we have that problem on most of our headers?  I can't remember it biting us in practice, but it's something to think about how we'd respond to it.  We can preemptively throw away all our case classes just in case, or give a new version some awful "v2" kind of name.  I think that's a wider discussion than this header, though.

Reply:
Wait a minute.. But isn't the whole header just a single value, is it? So, wouldn't it be better to implement it in a simple single-value ADT pattern, like this:
```scala
sealed trait `X-Content-Type-Options`

object `X-Content-Type-Options` {
  case object nosniff extends `X-Content-Type-Options`
}
```
What I mean is that it looks like it does not make any sense to keep a string value within the header instance. I.e. the "nosniff" string has to be parsed, of course, but there's no need to keep in the parsed instance. Just because it is always going to be "nosniff" anyway, no matter what.

Reply:
Thank you for feedback.
This can also be done in [`Access-Control-Allow-Credentials`][ACAC], since this header can accept only `true` value.

[ACAC]: https://github.com/http4s/http4s/blob/series/0.23/core/shared/src/main/scala/org/http4s/headers/Access-Control-Allow-Credentials.scala

Reply:
That's an ideal model for today's spec, but what if they add a second valid value a year from now?  Adding cases to a sealed trait breaks compatibility.  The basic tradeoff is do we want a well-designed ADT for the present spec, or do we want a less precise model that can support evolution of the spec?

I think, for 0.23, the precise model is good.  The futureproofing of headers is a good thought experiment to throw onto the 1.0 pile.  I really don't know how I feel about it without studying how often these specs change _in practice_.

Reply:
That's a fair concern. We could allow us a bit more room for future changes, if we hid the implementation type behind a regular `val`, i.e.:
```scala
object `X-Content-Type-Options` {
  val nosniff: `X-Content-Type-Options` = Nosniff
  
  private case object Nosniff extends `X-Content-Type-Options`
}
```
I guess, that would prevent the `nosniff` object type from being in the bin-compat scope.

That said, I don't really think it is worth the effort – the `X-Content-Type-Options` header has not changed since it was first introduced many years ago. So no signs it is going to change any time soon.

Reply:
One more thing to think about (a minor one though) – what do we expect to get as a result of its `toString` call? Currently it will be just `nosniff` string – is that what we actually expect? Or should it be overridden with `"X-Content-Type-Options(nosniff)"` or something like that?